sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuInsertIntoHadoopFsRelationCommand.scala

```diff
@@ -121,7 +121,7 @@ case class GpuInsertIntoHadoopFsRelationCommand(
       val pathExists = fs.exists(qualifiedOutputPath)
       (mode, pathExists) match {
         case (SaveMode.ErrorIfExists, true) =>
-          throw new AnalysisException(s"path $qualifiedOutputPath already exists.")
+          throw RapidsErrorUtils.dataPathNotExistError(qualifiedOutputPath.toString)
```

Comment:
This is the wrong error message to throw for this error.  The error message will state that it does _not_ exist, but instead it _does_exist, and that's the problem.  RapidsQueryErrorUtils.outputPathAlreadyExistsError would be more appropriate.  Note that the error being thrown here in Spark 4.0.0 is via QueryExecutionErrors.saveModeUnsupportedError.

Reply:
Thanks. 

> Note that the error being thrown here in Spark 4.0.0 is via QueryExecutionErrors.saveModeUnsupportedError.

I am not sure what you mean here.

Reply:
Just noting that there's a new interface in Spark 4.0.0 for throwing this type of error, see https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala#L139.  I think it's probably fine to use the outputPathAlreadyExistsError, but technically Spark 4.0.0 is using a different error utility method here.